pvanalysis/_pvanalysis.py

```diff
@@ -1152,15 +1143,13 @@ def plot_fitresult(self, vlim: list = [0, 1e10],
         if len(self.popt['ridge'][0]) == 5:
             self.avevsys = (self.popt['edge'][0][4]
                             + self.popt['ridge'][0][4]) / 2.
-            self.vsys += self.avevsys
-            self.popt['edge'][0][4] -= self.avevsys
-            self.popt['ridge'][0][4] -= self.avevsys
+            self.vsys_opt = self.vsys + self.avevsys
         else:
             self.avevsys = 0
```

Comment:
self.vsys_opt is not defined in the case of "else". So, self.avevsys = 0 --> self.vsys_opt = self.vsys?